packages/@react-aria/utils/test/mergeProps.test.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import classNames from 'classnames';
+import {mergeIds} from '../src/useId';
+import {mergeProps} from '../';
+
+
+describe('mergeProps', function () {
+  it('handles one argument', function () {
+    let onClick = () => {};
+    let classname = 'primary';
+    let id = 'test_id';
+    let mergedProps = mergeProps({onClick, classname, id});
+    expect(mergedProps.onClick).toBe(onClick);
+    expect(mergedProps.classname).toBe(classname);
+    expect(mergedProps.id).toBe(id);
+  });
+
+  it('combines callbacks', function () {
+    console.log = jest.fn();
+    let message1 = 'click1';
+    let message2 = 'click2';
+    let message3 = 'click3';
+
+    let a = {onClick: () => console.log(message1)};
+    let b = {onClick: () => console.log(message2)};
+    let c = {onClick: () => console.log(message3)};
+    let mergedProps = mergeProps(a, b, c);
+    mergedProps.onClick();
+    expect(console.log).toHaveBeenCalledWith(message1);
+    expect(console.log).toHaveBeenCalledWith(message2);
+    expect(console.log).toHaveBeenCalledWith(message3);
+    expect(console.log).toHaveBeenCalledTimes(3);
+  });
+
+  it('merges props with different keys', function () {
+    console.log = jest.fn();
+    let click1 = 'click1';
+    let click2 = 'click2';
+    let hover = 'hover';
+    let focus = 'focus';
+    let margin = 2;
+    const mergedProps = mergeProps(
+      {onClick: () => console.log(click1)},
+      {onHover: () => console.log(hover), styles: {margin}},
+      {onClick: () => console.log(click2), onFocus: () => console.log(focus)}
+    );
+    mergedProps.onClick();
+    mergedProps.onFocus();
+    mergedProps.onHover();
+    expect(console.log).toHaveBeenCalledWith(click1);
+    expect(console.log).toHaveBeenCalledWith(click2);
+    expect(console.log).toHaveBeenCalledWith(hover);
+    expect(console.log).toHaveBeenCalledWith(focus);
+    expect(console.log).toHaveBeenCalledTimes(4);
+    expect(mergedProps.styles.margin).toBe(margin);
+  });
+
+  it('combines classNames', function () {
+    let className1 = 'primary';
+    let className2 = 'hover';
+    let className3 = 'hover';
```

Comment:
this raises an interesting point, should we be deduping classnames?
there's nothing particularly wrong with multiple of the same that I'm aware of, so I'm inclined to leave it alone

Reply:
Ah, that's funny, this was actually a mistake in my test, but you're right, I think just letting the `classnames` library do what it does is probably the best move.